lib/ansible/modules/database/aerospike/aerospike_migrations.py

```diff
@@ -0,0 +1,535 @@
+    LIB_FOUND = False
+else:
+    LIB_FOUND = True
+
+
+def run_module():
+    """run ansible module"""
+    module_args = dict(
+        host=dict(type='str', required=False, default='localhost'),
+        port=dict(type='int', required=False, default=3000),
+        connect_timeout=dict(type='int', required=False, default=1000),
+        consecutive_good_checks=dict(type='int', required=False, default=3),
+        sleep_between_checks=dict(type='int', required=False, default=60),
+        tries_limit=dict(type='int', requires=False, default=300),
+        local_only=dict(type='bool', required=True),
+        min_cluster_size=dict(type='int', required=False, default=1),
+        target_cluster_size=dict(type='int', required=False, default=None),
+        fail_on_cluster_change=dict(type='bool', required=False, default=True),
+        migrate_tx_key=dict(type='str', required=False,
+                            default="migrate_tx_partitions_remaining"),
+        migrate_rx_key=dict(type='str', required=False,
+                            default="migrate_rx_partitions_remaining")
+    )
+
+    result = dict(
+        changed=False,
+        original_message='',
+        message=''
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+    if not LIB_FOUND:
+        module.fail_json(
+            msg="A required module was not found. This playbook" +
+            " requires the 'aerospike' 'time' and 're' modules. " +
+            "Please run 'pip install aerospike'. The other modules" +
+            " should be included in a basic python install."
+        )
+
+    if module.check_mode:
+        return result
+
+    migrations = Migrations(module)
+    # migrations.has_migs accepts 1 argument,
+    # a bool to specify checking local node only or entire cluster
+    has_migrations, skip_reason = migrations.has_migs(module.params['local_only'])
+    if not has_migrations:
+        result['message'] = "No migrations"
+    else:
+        result['message'] = skip_reason
+        module.fail_json(
+            msg="Failed.", **result
+        )
+
+    module.exit_json(**result)
+
+
+class Migrations:
+    """ Check or wait for migrations between nodes """
+
+    def __init__(self, module):
+        self.module = module
+        self._client = self._create_client().connect()
+        self._nodes = {}
+        self._update_nodes_list()
+        self._cluster_statistics = {}
+        self._update_cluster_statistics()
+        self._namespaces = set()
+        self._update_cluster_namespace_list()
+        self._build_list = set()
+        self._update_build_list()
+        self._start_cluster_key = \
+            self._cluster_statistics[self._nodes[0]]['cluster_key']
+
+    # TODO: add support for auth, tls, and other special features
+    # I won't use those features, so I'll wait until somebody complains
+    # or does it for me (Cross fingers)
+    # create the client object
```

Comment:
Python uses [docstrings](https://en.wikipedia.org/wiki/Docstring#Python) for method documentation, not comments.